picard/ui/matchdetailsdialog/matchdetails.py

```diff
@@ -0,0 +1,211 @@
+    current_file = ""
+
+    options = [
+        config.Option("persist", dialog_header_state, QtCore.QByteArray())
+    ]
+
+    def __init__(self, parent):
+        """ Initialize the dialog box object with information about itself, including the columns in the table/grid."""
+        super().__init__(parent)
+        self.file_ = None
+        self.setWindowTitle(_("Match Details"))
+        self.columns = [
+            ('nbr',                  "_Log\nSeq"),
+            ('match_dtl_entry_type', "_Entry Type"),
+            ('is_new',               "_Better?"),
+            ('rec_id',               "_Recording ID"),
+            ('rec_title',            "Recording Title"),
+            ('rec_len',              "Recording\nLength"),
+            ('rec_artist',           "Recording\nArtist"),
+            ('rg_id',                "_Release Group ID"),
+            ('rg_title',             "Release Group\nTitle"),
+            ('rg_pri_type',          "Release Group\nPri Type"),
+            ('rg_sec_type',          "Release Group\nSec Types"),
+            ('rel_id',               "_Release ID"),
+            ('rel_ctry',             "Release\nCountry"),
+            ('rel_dt',               "Release\nDate"),
+            ('med_fmt',              "Medium\nFormat"),
+            ('med_pos',              "Medium\nPosition"),
+            ('med_tot_cnt',          "Medium\nCount"),
+            ('trk_id',               "_Track ID"),
+            ('trk_pos',              "Track\nPosition"),
+            ('med_trk_cnt',          "Medium\nTrack\nCount"),
+            ('rel_tot_trk_cnt',      "Release\nTotal Track\nCount"),
+            ('parts',                "_Scoring Parts"),
+            ('release_parts',        "_Scoring Release Parts"),
+            ('recording_base_score', "_Recording\nBase Score\n(Sources Ratio)"),
+            ('acoust_id',            "AcoustId ID"),
+            ('rec_src_cnt',          "AcoustId\nRecording\nSource\nCount"),
+            ('similarity',           "Similarity")
+        ]
+
+    def load_match_details(self, track, file_):
+        """ Retrieves data and loads it to the screen/dialog box """
+        if file_ is not None:
+            # NOTE: some of the values, like meta, are not exposed as constants
+            fp = file_.acoustid_fingerprint if hasattr(file_, 'acoustid_fingerprint') else ""
+            dur = file_.acoustid_length if hasattr(file_, 'acoustid_length') else ""
+
+            fingerprint_lkp_url = ("https://" + ACOUSTID_HOST + "/v2/lookup"
+                                   + "?meta=recordings%20releasegroups%20releases%20tracks%20compress%20sources"
+                                   + "&fingerprint=" + fp
+                                   + "&duration=" + str(dur)
+                                   + "&client=" + ACOUSTID_KEY
+                                   + "&clientversion=" + PICARD_VERSION_STR
+                                   + "&format=json") if fp is not None else ""
```

Comment:
I'd make this test more obvious. If I understand correctly you set `fingerprint_lkp_url` to empty string if `fp` is None. Since the expression is already very long, I almost missed it.

But in this case, you still call https://github.com/metabrainz/picard/pull/1497/files#diff-a470bf968b12bbf5662c8ba4c21f00f7R172 which builds a link with an empty url.

1. I'd make the test `fp is None` more obvious
2. Get rid of `set_acoustid_lookup_url_edit` which is only used here
3. Better handling of the `fp is None` when it comes to `acoustid_lookup_url` label (hide?)

Reply:
Done. all cleaned up, `fp is None` is more prominent, and the clickable link in the UI is only shown if the fingerprint is available.  also expanded the fp abbreviation :)